presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -0,0 +1,281 @@
+
+            ImmutableMap.Builder<ColumnHandle, ColumnStatistics> columnHandleBuilder = ImmutableMap.builder();
+            double recordCount = summary.getRecordCount();
+            for (IcebergColumnHandle columnHandle : idToColumnHandle.values()) {
+                int fieldId = columnHandle.getId();
+                ColumnStatistics.Builder columnBuilder = new ColumnStatistics.Builder();
+                Long nullCount = summary.getNullCounts().get(fieldId);
+                if (nullCount != null) {
+                    columnBuilder.setNullsFraction(Estimate.of(nullCount / recordCount));
+                }
+                if (summary.getColumnSizes() != null) {
+                    Long columnSize = summary.getColumnSizes().get(fieldId);
+                    if (columnSize != null) {
+                        columnBuilder.setDataSize(Estimate.of(columnSize));
+                    }
+                }
+                Object min = summary.getMinValues().get(fieldId);
+                Object max = summary.getMaxValues().get(fieldId);
+                if (min instanceof Number && max instanceof Number) {
+                    columnBuilder.setRange(Optional.of(new DoubleRange(((Number) min).doubleValue(), ((Number) max).doubleValue())));
+                }
+                columnHandleBuilder.put(columnHandle, columnBuilder.build());
+            }
+            return new TableStatistics(Estimate.of(recordCount), columnHandleBuilder.build());
+        }
+    }
+
+    private boolean partitionMatches(
+            Partition partition,
+            Constraint constraint,
+            List<PartitionField> partitionFields,
+            Map<Integer, ColumnFieldDetails> fieldDetails)
+    {
+        TupleDomain<ColumnHandle> constraintSummary = constraint.getSummary();
+
+        Map<ColumnHandle, Domain> domains = constraintSummary.getDomains().get();
+
+        Predicate<Map<ColumnHandle, NullableValue>> predicate = constraint.predicate().orElse(value -> true);
+
+        ImmutableMap.Builder<ColumnHandle, NullableValue> nullableValueBuilder = ImmutableMap.builder();
+
+        for (int index = 0; index < partitionFields.size(); index++) {
+            PartitionField field = partitionFields.get(index);
+            int fieldId = field.sourceId();
+            ColumnFieldDetails details = fieldDetails.get(fieldId);
+            IcebergColumnHandle column = details.getColumnHandle();
+            Object value = PartitionTable.convert(partition.getValues().get(index, details.getJavaClass()), partition.getIdToTypeMapping().get(fieldId));
+            Domain allowedDomain = domains.get(column);
+            if (allowedDomain != null && !allowedDomain.includesNullableValue(value)) {
+                return false;
+            }
+            nullableValueBuilder.put(column, value == null ? NullableValue.asNull(details.getPrestoType())
+                    : HiveUtil.parsePartitionValue("unnamed", value.toString(), details.getPrestoType(), UTC));
```

Comment:
I think I get this point.  I'll take a swing at it.

Reply:
I did this conversion, but ran into the problem that the `Constraint` predicate can't be carried in the `TupleDomain` passed to `TableScan`, so I still needed most of the machinery to create the map of per-column-per-`DataFile` `NullableValues` needed by the `Constraint` predicate :(

Reply:
We should fork and simplify this code, since the Hive code will return return the wrong result for strings equaling `__HIVE_DEFAULT_PARTITION__`, plus the conversion logic is more lax than we need for Iceberg. For example, the Hive code handles multiple representations for boolean values, etc.

For this, the conversion can go directly from the `value` object to `NullableValue`. No need to round trip through a string. i.e. if the Iceberg object is a `Long`, we just call `NullableValue.of(...)` directly.

We want to avoid using Hive code that is "close enough" or that just happens to be the same (but might change in the future) for Iceberg.